src/high-level/matrix.lisp

```diff
@@ -219,6 +219,11 @@ ELEMENT-TYPE, CAST, COPY-TENSOR, DEEP-COPY-TENSOR, TREF, SETF TREF)"
   "Whether MATRIX is a unitary matrix"
   (= matrix (conjugate-transpose matrix) epsilon))
 
+(defun block-matrix-p (matrix)
```

Comment:
Note: the general usage is that a "block matrix" is a matrix with an associated decomposition, e.g.
```
[[A, B, C],
  [D, E, F]]
```
would be a valid block decomposition if `nrows(A) = nrows(B) = nrows(C)`, `ncols(A) = ncols(D)`, and so on. In this usage, the matrix doesn't have to be square. I'd suggest changing the name of your defun and adding a docstring if you want to keep the current behavior.